consensus/ibft/fork/helper.go

```diff
@@ -0,0 +1,91 @@
+package fork
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+
+	"github.com/0xPolygon/polygon-edge/validators/valset"
+	"github.com/0xPolygon/polygon-edge/validators/valset/snapshot"
+	"github.com/hashicorp/go-hclog"
+)
+
+func ibftTypeToSourceType(ibftType IBFTType) valset.SourceType {
+	switch ibftType {
+	case PoA:
+		return valset.Snapshot
+	case PoS:
+		return valset.Contract
+	default:
+		return valset.Snapshot
+	}
+}
+
+func loadSnapshotMetadata(
+	logger hclog.Logger,
+	path string,
+) *snapshot.SnapshotMetadata {
+	var meta *snapshot.SnapshotMetadata
+	if err := readDataStore(path, &meta); err != nil {
+		// if we can't read metadata file delete it
+		// and log the error that we've encountered
+		logger.Error("Could not read metadata snapshot store file", "err", err.Error())
+		os.Remove(path)
```

Comment:
Why don't you check for errors here?

Reply:
Add error handling at d69c6b33c5533ec0ac56ec4d5aff48ea0f369b14